main.tf

```
@@ -16,18 +16,30 @@ locals {
     length(var.iam_role_name) > 0 ? var.iam_role_name : "${var.prefix}-iam-${random_id.uniq.hex}"
   )
   mfa_delete = var.bucket_enable_versioning && var.bucket_enable_mfa_delete
+  bucket_encryption_enabled = var.bucket_enable_encryption && var.bucket_encryption_enabled
+  bucket_logs_enabled = var.bucket_logs_enabled && var.bucket_enable_logs
+  bucket_versioning_enabled = var.bucket_enable_versioning && var.bucket_versioning_enabled
 }
 
 resource "random_id" "uniq" {
   byte_length = 4
 }
 
+resource "aws_kms_key" "lacework_kms_key" {
+  count                   = var.bucket_sse_key_arn == "" ? 1 : 0
+  description             = "lacework_cloudtrail_kms_key"
```

Maybe something a little more descriptive here.
```suggestion
  description             = "A KMS key used to encrypt CloudTrail logs which are monitored by Lacework"
```